src/scripts/util.sh

```diff
@@ -37,31 +37,282 @@ error() {
+
+        # The resources we're looking for, that we will reset between server blocks
+        local cert_names=()
+        local server_names=()
+
+        local lineno=0
+        while IFS="" read -r line || [ -n "$line" ]; do
+            lineno=$((lineno + 1))
+            if [[ "$line" =~ ^([[:space:]]*)server[[:space:]]*\{ ]]; then
+                if [ $in_server_block -eq 1 ]; then
+                    error "Matching server block while already in a server block ($conf_file:$lineno)"
+                    # Do we want to go to next file, exit or keep parsing ?
+                fi
+                # We entered a server block, so we'll need to start checking what's up
+                in_server_block=1
+                leader_server_block_spaces=${BASH_REMATCH[1]}
+            elif [ $in_server_block -eq 0 ]; then
+                # We are not in a server block, nothing for us to do here, let's skip
+                continue
+            elif [[ "$line" =~ ^${leading_server_block_spaces}\} ]]; then
+                # We add the found certificate names and the corresponding server names,
+                # in case it already exists, this will be appended nicely as if it was
+                # all in the same server block. In case there was no 'ssl_certificate_key'
+                # entry, we can simply ignore it nicely.
+                if [ ${#server_names[@]} -eq 0 ]; then
+                    error "Missing 'server_name' for server block ($conf_file:$lineno)"
+                elif [ ${#cert_names[@]} -gt 0 ]; then
+                    for cert_name in "${cert_names[@]}"; do
+                        certs["$cert_name"]="${certs["$cert_name"]}${certs["$cert_name"]:+ }${server_names[@]}"
+                    done
+                fi
+
+                # We are leaving a server block, so we can clean-up behind us
+                in_server_block=0
+                leadin_server_block_spaces=
+
+                cert_names=()
+                server_names=()
+            elif [[ "$line" =~ ^[[:space:]]*ssl_certificate_key[[:space:]]*/etc/letsencrypt/live/([^;]*)/privkey.pem[[:space:]]*\; ]]; then
+                cert_names+=(${BASH_REMATCH[1]})
+            elif [[ "$line" =~ ^[[:space:]]*server_name[[:space:]]*([^\;]*)[[:space:]]*\; ]] ||\
+                    [[ "$line" =~ ^[[:space:]]*#[[:space:]]*cert:add_domains[[:space:]]*(.*)[[:space:]]*$ ]]; then
+                for server_name in ${BASH_REMATCH[1]}; do
+                    if is_ip "$server_name"; then
+                        # This is matching an IP, we can decide to keep them (local CA) or ignore them (public CA)
+                        if [ "$skip_ips" == "skip_ips" ]; then
+                            debug "IP address '${server_name}' ignored as server_name ($conf_file:$lineno)"
+                            continue
+                        fi
+                    elif [[ ! "$server_name" =~ ^(\*|[-_[:alnum:]]+)(\.[-_[:alnum:]]+)+$ ]]; then
```

Comment:
Just realized I might need a slight adjustment here to support a minimum ndot, since for local CAs we might want to consider certs for localhost or whatever. For LE we will most likely want to skip those however.